libs/blockscout-service-launcher/src/launch.rs

```diff
@@ -63,20 +70,31 @@ where
     res?
 }
 
-fn http_serve<R>(http: R, metrics: PrometheusMetrics, addr: SocketAddr) -> actix_web::dev::Server
+fn http_serve<R>(
+    http: R,
+    metrics: Option<PrometheusMetrics>,
+    addr: SocketAddr,
+) -> actix_web::dev::Server
 where
     R: HttpRouter + Send + Sync + Clone + 'static,
 {
     tracing::info!("starting http server on addr {}", addr);
-    let server = HttpServer::new(move || {
-        App::new()
-            .wrap(metrics.clone())
-            .configure(configure_router(&http))
-    })
-    .bind(addr)
-    .unwrap_or_else(|_| panic!("failed to bind server"));
 
-    server.run()
+    if let Some(metrics) = metrics {
+        HttpServer::new(move || {
```

Comment:
make if inside `move ||` lambda

Reply:
You cannot do that. `App::new().wrap(metrics.clone()).configure(configure_router(&http))` and `App::new().configure(configure_router(&http))` are of different types and result in different http servers